Added travis hook

jrmlhermitte opened the pull request:
Adding a travis hook for testing. Could be useful.
Right now just tests that library installs correctly. No tests have been written.
Should talk to @yugangzhang whether or not he wants to add tests before deciding to merge.

mrakitin commented on test-requirements.txt:
```diff
@@ -0,0 +1 @@
+coverage
```
Should we also add `ipython` dependency? Travis fails with:
```
pyCHX/tests/test_import.py::test_import FAILED                           [100%]
=================================== FAILURES ===================================
_________________________________ test_import __________________________________
    def test_import():
        # test handler import
        # this is mainly an example
>       from pyCHX.chx_libs import EigerHandler
pyCHX/tests/test_import.py:4: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
    """
    
>   from IPython.core.magics.display import Javascript
E   ModuleNotFoundError: No module named 'IPython'
pyCHX/chx_libs.py:7: ModuleNotFoundError
=========================== 1 failed in 0.19 seconds ===========================
```

jrmlhermitte replied:
yeah unfortunately :-(
I ran this locally, found a few other issues. compiles locally, now waiting for Travis.
when it compiles ill summarize (and point out whiat issues we need to raise here)

mrakitin commented:
Looks great now!

jrmlhermitte commented on .travis.yml:
```diff
@@ -0,0 +1,24 @@
+  - pip install -r test-requirements.txt
+    # make a fake instance of databroker (so imports work)
+  - mkdir -p ~/.config/databroker
+  - cp extra/_legacy_config.yml ~/.config/databroker/_legacy_config.yml
```
I wanted to point this out to @danielballan . We added tests to `pyCHX` which required adding these two configs.
the first config is a sign that we need to remove legacy dependencies (new version of databroker will search for `_legacy_config.yml` and only create `DataBroker` within its namespace if it exists, so that legacy `from databroker import DataBroker` can still work)

jrmlhermitte commented on extra/_legacy_config.yml:
```diff
@@ -0,0 +1,16 @@
+# dummy file
+metadatastore:
+    module: databroker.headersource.mongo
```
again @danielballan this is a test config file only for purposes of testing chx's import libraries.

jrmlhermitte commented on requirements.txt:
```diff
@@ -1,17 +1,18 @@
-cython
+numpy
```
Don't mind the swap between `numpy` and `cython` ordering. I encountered an error which I thought was due to import precedence (but I was just running the install incorrectly).

jrmlhermitte commented:
Great thanks for merging @mrakitin !

Let me now summarize what this PR was doing, so that we can run it by @danielballan  and he's aware of the changes (in case he might have a better suggestion)
1. We added a `.travis` file and activated a travis hook to do automated testing
2. We added just one simple test:
`from pyCHX.chx_libs import EigerHandler`
With all the dependencies required in this one statement this seems sufficient to test the imports/dependencies for now. More should be added later. I also encourage @yugangzhang to add unit tests to his functions if he can (and we're happy to help guide you).
3. Added `tqdm` and `ipython` to the requirements file. Also,
4. This uncovered a `from filestore.path_only_handlers import RawHandler` lying around.

Anyway, I think this is a useful addition. Let me know if you have any comments.

danielballan commented:
Wonderful. Yes, additional test from BL staff would be helpful. Did you grep for `filestore` to be sure we caught them all?

mrakitin commented:
https://github.com/NSLS-II-CHX/pyCHX/blob/master/pyCHX/Spatial_Correlation_Function.py#L2 still contains it.

mrakitin commented:
Interesting... It's imported, but is not used. Let's just remove the line:
```py
from filestore.api import register_handler, deregister_handler
```

mrakitin commented:
https://github.com/NSLS-II-CHX/pyCHX/pull/11 should fix it.
